doc/example/synthetic_data.ipynb

```diff
@@ -0,0 +1,258 @@
+{
```

Comment:
Can you maybe also give short comments (as `.md` cells) between the many cells, that you have. One sentence is probably enough per cell, but helps the reader to understand what you are doing there :)

Reply:
Thanks, added some additional descriptions.